docs: external dependencies and resource weight (#4596)
werf/werf: docs/pages_en/advanced/helm/deploy_process/steps.md

When running the `werf converge` command, werf starts the deployment process that includes the following steps:

> 1. Render chart templates into a single list of Kubernetes resource manifests and validate them.
> 2. Run `pre-install` or `pre-upgrade` [hooks]({{ "/advanced/helm/deploy_process/helm_hooks.html" | true_relative_url }}) sequentially, sorted by their weight, and track each hook until its termination while printing its logs in the process.
> 3. Group non-hook Kubernetes resources by their [weight]({{ "/advanced/helm/deploy_process/deployment_order.html" | true_relative_url }}) and deploy each group sequentially, sorted by their weight: create/update/delete resources and track these resources until they are ready while printing logs in the process.
> 4. Run `post-install` or `post-upgrade` [hooks]({{ "/advanced/helm/deploy_process/helm_hooks.html" | true_relative_url }}) the same way as with the `pre-install` and `pre-upgrade` hooks.

Reviewer: ```suggestion
1. Rendering chart templates into a single list of Kubernetes resource manifests and validating them.
2. Sequentially running `pre-install` or `pre-upgrade` [hooks]({{ "/advanced/helm/deploy_process/helm_hooks.html" | true_relative_url }}) sorted by their weight and tracking each hook to completion while printing its logs in the process.
3. Grouping non-hook Kubernetes resources by their [weight]({{ "/advanced/helm/deploy_process/deployment_order.html" | true_relative_url }}) and deploying each group sequentially according to its weight: creating/updating/deleting resources and tracking them until they are ready while printing logs in the process.
4. Running `post-install` or `post-upgrade` [hooks]({{ "/advanced/helm/deploy_process/helm_hooks.html" | true_relative_url }}) the same way as `pre-install` and `pre-upgrade` hooks.
```

## Resource tracking

> werf uses the [kubedog library](https://github.com/werf/kubedog) to track resources. Tracking behaviour can be configured for each resource using [resource annotations]({{ "/reference/deploy_annotations.html" | true_relative_url }}), which should be set in the chart templates.

Reviewer: ```suggestion
werf uses the [kubedog library](https://github.com/werf/kubedog) to track resources. Tracking behavior can be configured for each resource by setting [resource annotations]({{ "/reference/deploy_annotations.html" | true_relative_url }}) in the chart templates.
```
